R/module_data_summary.R

```diff
@@ -163,83 +178,86 @@ get_filter_overview <- function(teal_data) {
       } else {
         joinkeys[dataname, dataname]
       }
-      get_object_filter_overview(
-        filtered_data = filtered_data_objs[[dataname]],
-        unfiltered_data = unfiltered_data_objs[[dataname]],
+      get_filter_overview(
+        current_data = current_data_objs[[dataname]],
+        initial_data = initial_data_objs[[dataname]],
         dataname = dataname,
         subject_keys = subject_keys
       )
     }
   )
 
-  unssuported_idx <- vapply(rows, function(x) all(is.na(x[-1])), logical(1)) # this is mainly for vectors
-  do.call(rbind, c(rows[!unssuported_idx], rows[unssuported_idx]))
+  do.call(.smart_rbind, out)
```

Comment:
`.smart_rbind` because `get_filter_overview` output is not restricted. Previously every method have returned: 
- `dataname`, 
- `obs` and `obs_filtered`: number of filtered and unfiltered rows which were formatted into single column in `renderUI` to `Obs = <filtered>/<unfiltered>`
- `subjects` and `subjects_filtered`

Now methods can return any columns but formatting occurs in the method and not in the `renderUI`
- `dataname` 
- `obs`: formatted counts `"<filtered>/<unfiltered>"`
- `subjects` 

I've modified the way what function returns to allow easy extension in the future. `.rbind_smart` is used only to bind overviews which might be different for different object types.